### PR TITLE
[v1.23] Bump 1.23.1

### PR DESCRIPTION
**Pull request by ashley-cui**
#### What type of PR is this?

> /kind api-change
> /kind bug
> /kind cleanup
> /kind deprecation
> /kind design
> /kind documentation
> /kind failing-test 
> /kind feature
> /kind flake
> /kind other

#### Does this PR introduce a user-facing change?

```release-note

```

### CODE REVIEW

**Comment by ashley-cui** [comment]
@TomSweeneyRedHat @containers/buildah-maintainers PTAL (extra carefully, as I've never done a buildah release before :smiling_face_with_tear: )

File: contrib/rpm/buildah.spec [hunk 1]
```diff
@@ -100,6 +100,8 @@ make DESTDIR=%{buildroot} PREFIX=%{_prefix} install install.completions
 %{_datadir}/bash-completion/completions/*
 
 %changelog
+* Mon Sep 27, 2021 Ashley Cui <acui@redhat.com> 1.23.2-dev-1
```
[the remark this answers is not in the document]
Comment (nalind) [reply contrib/rpm/buildah.spec:103]: Yes, historically we only bother with dev tags on the main branch, so the 1.23.1-1 changelog entry below it is sufficient.  Ditto for `define/types.go`'s `Version` constant.

[the remark this answers is not in the document]
Comment (ashley-cui) [reply contrib/rpm/buildah.spec]: Okay I dropped the -dev commit, PTAL again, thanks! :)

**Comment by nalind** [comment]
LGTM

**Comment by TomSweeneyRedHat** [comment]
LGTM
/hold 
# for tests
Thx and nice job @ashley-cui !

**Pull request merged**